src/Razor/test/Microsoft.AspNetCore.Razor.VSCode.Grammar.Test/tests/__snapshots__/GrammarTests.test.ts.snap

```diff
@@ -5184,22 +5184,22 @@ Line:
 
 Line:         <text>This is not a special transition tag</text>
  - token from 0 to 8 (        ) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs
- - token from 8 to 9 (<) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, punctuation.definition.tag.begin.html
- - token from 9 to 13 (text) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, entity.name.tag.html
- - token from 13 to 14 (>) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, punctuation.definition.tag.end.html
+ - token from 8 to 14 (<text>) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, keyword.control.cshtml.transition.textTag.open
  - token from 14 to 50 (This is not a special transition tag) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs
- - token from 50 to 52 (</) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, punctuation.definition.tag.begin.html
- - token from 52 to 56 (text) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, entity.name.tag.html
- - token from 56 to 57 (>) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, punctuation.definition.tag.end.html
+ - token from 50 to 57 (</text>) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, keyword.control.cshtml.transition.textTag.close
 
 Line:         Hello World
- - token from 0 to 20 (        Hello World) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs
+ - token from 0 to 8 (        ) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs
+ - token from 8 to 13 (Hello) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, variable.other.readwrite.cs
+ - token from 13 to 14 ( ) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs
+ - token from 14 to 19 (World) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, variable.other.readwrite.cs
 
 Line:     </p>
  - token from 0 to 4 (    ) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs
- - token from 4 to 6 (</) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, punctuation.definition.tag.begin.html
- - token from 6 to 7 (p) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, entity.name.tag.html
- - token from 7 to 8 (>) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, punctuation.definition.tag.end.html
+ - token from 4 to 5 (<) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, keyword.operator.relational.cs
+ - token from 5 to 6 (/) with scopes text.aspnetcorerazor, meta.structure.razor.codeblock, source.cs, keyword.operator.arithmetic.cs
```

Comment:
Yeah, I'm not exactly sure what's going on here, but my guess is it's not able to end one of the `<strong>` tags.

Here's a question - is the code block _inside_ the beginning strong tag on L15 valid Razor?

Reply:
All right, added back the back-reference so that we only capture matching tags. (I'm kind of amazed that the back-reference works, can't imagine how the engine would try to parse back-references if the end pattern _also_ had 2 captures by that point.)